### PR TITLE
Implement handling N_In

### PR DESCRIPTION
**Pull request by xbauch**
That is the check that a value `x` is inside a range `<low,high>`. The implementation simply builds a conjunction `x \geq low /\ x \leq high`. Re-enables existing test (changed so that the assertion is failing).

### CODE REVIEW

File: experiments/golden-results/UKNI-Information-Barrier-summary.txt [hunk 1]
```diff
@@ -1,13 +1,8 @@
-Occurs: 47 times
+Occurs: 46 times
```
Comment (martin-cs) [inline experiments/golden-results/UKNI-Information-Barrier-summary.txt]: I'm obviously happy to see these going down but do you see why this change reduces the occurrences of this error?

File: gnat2goto/driver/tree_walk.adb [hunk 2]
```diff
@@ -1419,6 +1417,28 @@ package body Tree_Walk is
       end case;
    end Do_Expression;
 
+   function Do_In (N : Node_Id) return Irep is
+      Left_Op : constant Irep := Do_Expression (Left_Opnd (N));
```
Comment (martin-cs) [inline gnat2goto/driver/tree_walk.adb]: Does GNAT guarantee that this node will have lower and upper bounds like this?  For example with subtype inclusion?

Comment (xbauch) [reply gnat2goto/driver/tree_walk.adb]: I'm not entirely sure. I've added a check (to `Do_Expression`) that reports as unsupported feature if we ended up calling `Do_In` with a node that doesn't have the bounds.
